### PR TITLE
feat(tasks): add owner-private Nostr task events

### PR DESCRIPTION
**Pull request by FurnitAI**
## Summary

- add signed Buzz Tasks v1 kinds and strict payload/envelope validation
- keep public reads on the existing NIP-98 `/query` and NIP-42 Nostr surfaces; the `buzz_tasks` table is internal derived state only
- add atomic projection transitions plus kind-5/kind-9005 deletion replay
- use the existing `(created_at DESC, id ASC)` bridge keyset cursor and cover create/update/resolve mutations between pages
- p-gate task content, apply result-level owner gates, and deny task kinds in global FTS even on brownfield search vectors
- keep only canonical `community_id`, `channel_id`, and source Nostr event identity; construct `buzz://message?channel=<uuid>&id=<event-id>` after authorization
- add the agent-first client surface: `buzz-sdk` task builders and the `buzz tasks` CLI group (request/update/resolve/list)

## Review findings resolved

- **P1 Nostr-first:** removed the task-specific `GET /api/buzz-tasks` routes and DTO/list implementation. Clients use signed Nostr filters through `/query`; no new permanent HTTP task contract remains.
- **P1 deletion projection:** task event deletions now tombstone and rebuild derived state from remaining live signed events in the same transaction for both standard kind 5 and admin kind 9005 paths.
- **P2 pagination:** removed the OFFSET task cursor with the deleted list API. The documented public cursor is the bridge keyset over the complete deterministic `(created_at, id)` order.
- **P1 agent-first CLI:** kinds 44300–44302 now have the required `buzz-cli` command surface. `buzz-sdk` gains `TaskRef` + `build_task_requested/updated/resolved` (round-trip locked to `TaskEventV1::parse`), and `buzz tasks request|update|resolve|list` signs, re-validates through the exact task contract, and submits. The owner `p` tag defaults from the NIP-OA auth tag. Because task reads are owner-gated, `update`/`resolve` take the original request's signed identity explicitly and `list` is the owner-side `#p` view.
- **P2 duplicate requests:** an equal-version kind-44300 for an existing task identity with a differing payload now returns an error and rolls back the event insert instead of committing a second, unfoldable v1 request as `Stale`. Only a payload identical to the stored v1 projection remains `Stale`, mirroring the equal-version conflict rule in `require_open_newer`.

## Migration gate

Migration 0031 creates only the empty `buzz_tasks` projection and its constraints/index/write fence in one SQLx transaction with a 5-second `lock_timeout`. It does not alter `events.search_tsv`, rewrite `events`, or rebuild the GIN index. Lock behavior, interruption recovery, rollback, production-volume implications, and the separate staged FTS hardening plan are documented in `docs/buzz-tasks-migration-plan.md`.

## Verification

- `./bin/just ci`
- `./bin/just test`
- `cargo test -p buzz-core --test task_contract` — 6 passed
- `cargo test -p buzz-db --test task_projection -- --include-ignored --test-threads=1` — 8 passed against PostgreSQL (includes the new conflicting-duplicate rollback test)
- `cargo test -p buzz-sdk` — 258 passed (includes the task builder contract round-trip)
- `cargo test -p buzz-cli` — 344 passed (includes the tasks command registry and parser tests)
- targeted clippy with `-D warnings` — passed

## Scope boundary

PR 1 adds no task UI, reply/approval/choice/resolve API, preferences, EventKit/Reminders integration, or HTTPS/universal-link support. The HTTPS/universal-link security review remains a separate delivery before PR 2. Do not merge until the renewed review is complete.

### CODE REVIEW

File: crates/buzz-relay/src/router.rs [hunk 1]
```diff
@@ -72,6 +72,9 @@ pub fn build_router(state: Arc<AppState>) -> Router {
         .route("/events", post(api::bridge::submit_event))
         .route("/query", post(api::bridge::query_events))
         .route("/count", post(api::bridge::count_events))
+        // Buzz Tasks PR 1: owner-private, read-only projection surface.
+        .route("/api/buzz-tasks", get(api::tasks::list_tasks))
+        .route("/api/buzz-tasks/{task_id}", get(api::tasks::get_task))
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply crates/buzz-relay/src/router.rs]: **Fixed** in `fa1a8fc8d`. The task-specific GET routes and their list/detail DTOs were removed. Public task reads now use the existing signed Nostr bridge (`POST /query`) with kinds 44300-44302 and `#p=self`; the projection remains internal derived state only. NIP-BT documents the bridge contract and its `(created_at, id)` keyset cursor.

File: crates/buzz-db/src/task.rs [hunk 2]
```diff
@@ -0,0 +1,634 @@
...
+                t.due_at, t.status, t.source_created_at, t.source_version, \
+                t.source_updated_at, t.resolved_at \
+         FROM buzz_tasks t WHERE t.community_id=$1 AND t.assignee_pubkey=$2 AND t.id=$3 \
+           AND EXISTS (SELECT 1 FROM channel_members cm \
+                       WHERE cm.community_id=t.community_id AND cm.channel_id=t.channel_id \
+                         AND cm.pubkey=t.assignee_pubkey AND cm.removed_at IS NULL) \
+           AND EXISTS (SELECT 1 FROM events e \
+                       WHERE e.community_id=t.community_id AND e.id=t.source_event_id \
+                         AND e.channel_id=t.channel_id AND e.pubkey=t.agent_pubkey \
+                         AND e.deleted_at IS NULL)",
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply crates/buzz-db/src/task.rs]: **Fixed** in `fa1a8fc8d`. Both kind-5 and kind-9005 task deletions now call one transactional path that locks/tombstones the signed event and rebuilds (or removes) `buzz_tasks` from the remaining live stream before commit. The PostgreSQL test deletes resolved, updated, then requested events and proves reopen/restore/removal plus three tombstones.

File: crates/buzz-db/src/task.rs [hunk 3]
```diff
@@ -0,0 +1,634 @@
...
+    builder
+        .push(" ORDER BY (t.due_at IS NOT NULL AND t.due_at < ")
+        .push_bind(query.as_of)
+        .push(
+            ") DESC, CASE priority WHEN 'high' THEN 0 WHEN 'medium' THEN 1 ELSE 2 END ASC, \
+               t.due_at ASC NULLS LAST, t.source_created_at DESC, t.id ASC LIMIT ",
+        )
+        .push_bind(query.limit.clamp(1, 101))
+        .push(" OFFSET ")
+        .push_bind(query.offset.max(0));
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply crates/buzz-db/src/task.rs]: **Fixed** in `fa1a8fc8d`. Removing the task-specific read API also removed its OFFSET cursor. Task reads now use the bridge complete keyset `(created_at DESC, id ASC)` via `until + before_id`. The integration test forces same-second ties, creates a task, updates one, and resolves another between pages, then proves the original snapshot has no duplicate or missing event and that above-cursor mutations do not leak into page 2.

**Comment by FurnitAI** [comment]
@codex review

Please re-review commit `fa1a8fc8d`, including the Nostr-first read surface, full `(created_at, id)` keyset pagination test, transactional deletion replay, and migration lock/recovery plan. PR 2 remains out of scope.

File: crates/buzz-db/src/task.rs [hunk 4]
```diff
@@ -0,0 +1,677 @@
...
+            validate_identity(&existing, &parsed)?;
+            live.push((parsed, stored));
+        }
+        live.sort_by(|(left_task, left_event), (right_task, right_event)| {
+            left_task
+                .payload
+                .source_version()
+                .cmp(&right_task.payload.source_version())
+                .then_with(|| left_event.received_at.cmp(&right_event.received_at))
+                .then_with(|| left_event.event.id.cmp(&right_event.event.id))
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply crates/buzz-db/src/task.rs:566]: Fixed in 394520374. The projection now persists task_event_id and replay orders equal sourceVersion candidates with the persisted committed winner first, so deleting a different equal-version event cannot flip the projection. Added an ignored Postgres regression test covering this exact replay.

File: docs/nips/NIP-BT.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+fields are required together. Events created, updated, or resolved above that
+cursor belong to the live/head side and cannot duplicate or displace the
+remaining historical page.
+
+The bridge verifies NIP-98 against the exact host, `/query` path, HTTP method,
+and body, then applies the shared replay fence, admission limit, relay
+membership, exact `#p = authenticated owner`, current channel access, and a
+per-result owner gate. The WebSocket REQ form uses the same standard Nostr
+filter and NIP-42 identity, but the bridge is the pagination contract because
+standard NIP-01 filters do not carry `before_id`.
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply docs/nips/NIP-BT.md:159]: Fixed in 394520374. NIP-BT now documents two live filters: owner/channel-gated task kinds 44300–44302, plus a separate channel-access deletion filter for kind 5/9005. The client applies deletion only to an e-tag target already present in its local owner fold; deletion is an invalidation signal, not content or access.

**Comment by FurnitAI** [comment]
@codex review

**Comment by FurnitAI** [comment]
@codex review

File: crates/buzz-relay/src/handlers/event.rs [hunk 6]
```diff
@@ -454,12 +454,11 @@ async fn dispatch_persistent_event_inner(
...
     };
-    // For viewer-private events (kind:30622 DM visibility, kind:44200 agent turn
-    // metrics), live fan-out must reach only the owner — a kindless `ids:[…]`
+    // For viewer-private result-gated events, live fan-out must reach only the
+    // signed owner — a kindless `ids:[…]`
     // subscription can otherwise match it. Pull paths (HTTP /query, WS historical)
     // are gated separately by reader_authorized_for_event.
-    let owner_only_kind = kind_u32 == buzz_core::kind::KIND_DM_VISIBILITY
-        || kind_u32 == buzz_core::kind::KIND_AGENT_TURN_METRIC;
+    let owner_only_kind = buzz_core::kind::RESULT_GATED_KINDS.contains(&kind_u32);
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply crates/buzz-relay/src/handlers/event.rs:486]: Fixed in 150f8d9f6. Redis fan-out now applies the same per-recipient RESULT_GATED owner check as local ingest, after the shared tenant/channel access filter. This closes cross-node plaintext task delivery.

File: docs/nips/NIP-BT.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+  {
+    "kinds": [44300, 44301, 44302],
+    "#p": ["<authenticated owner pubkey>"],
+    "#h": ["<accessible channel UUID>"],
+    "limit": 50
+  },
+  {
+    "kinds": [5, 9005],
+    "#h": ["<accessible channel UUID>"],
+    "limit": 50
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply docs/nips/NIP-BT.md:130]: Fixed in 150f8d9f6. The relay already derives StoredEvent.channel_id from the kind-5 target, and the shared filters_match implementation falls back to that derived channel when a deletion has no signed h-tag. NIP-BT now explicitly documents this server-side derived-channel behavior and keeps deletions in a separate filter.

File: docs/nips/NIP-BT.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+The signed event stream is the source of truth. `buzz_tasks` is a rebuildable
+read projection keyed by `(community_id, task UUID)` with an additional unique
+source identity:
+
+```text
+(community_id, channel_id, source_event_id, assignee_pubkey)
+```
+
+Exact signed-event replay is a no-op. A transition with a lower or equal
+`sourceVersion` is stored for audit but does not change the projection. A newer
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply docs/nips/NIP-BT.md]: Fixed in 150f8d9f6. Conflicting equal sourceVersion transitions are now rejected and rolled back server-side. The public Nostr stream therefore has one reconstructible winner; the regression test proves no event or projection change is retained.

**Comment by FurnitAI** [comment]
@codex review

**Comment by FurnitAI** [comment]
@codex review

Please re-review commit `30caaa97a`. This round resolves both open findings:

- **P1 agent-first CLI:** `buzz-sdk` task builders (`TaskRef`, `build_task_requested/updated/resolved`, round-trip locked to `TaskEventV1::parse`) and the `buzz tasks` CLI group (request/update/resolve/list). Writes sign, re-validate through the exact task contract, then submit; the owner `p` tag defaults from the NIP-OA auth tag. Task reads are owner-gated on the relay, so `update`/`resolve` take the original request's signed identity explicitly and `list` is the owner-side `#p` view.
- **P2 duplicate requests:** equal-version kind-44300 with a differing payload now errors and rolls back the event insert; only an identical payload stays `Stale`. Covered by the new `conflicting_duplicate_request_rolls_back_and_identical_retry_stays_stale` projection test.

PR 2 remains out of scope.

File: crates/buzz-core/src/kind.rs [hunk 9]
```diff
@@ -544,6 +555,14 @@ pub const KIND_MEMBER_REMOVED_NOTIFICATION: u32 = 44101;
 /// See `docs/nips/NIP-AM.md`.
 pub const KIND_AGENT_TURN_METRIC: u32 = 44200;
 
+// Agent-to-owner task protocol (44300–44399)
+/// Buzz Tasks v1: an agent requests an action from its registered owner.
+pub const KIND_TASK_REQUESTED: u32 = 44300;
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply crates/buzz-core/src/kind.rs:560]: Addressed in 7b8325f90 (feat(tasks): add buzz tasks agent CLI surface). buzz-cli now ships the agent-first command surface for kinds 44300-44302: `buzz tasks request|update|resolve|list`, backed by new buzz-sdk TaskRef + build_task_requested/updated/resolved builders that serialize the buzz-core contract payloads. Writes re-validate through the exact task contract before submit. Since task reads are owner-gated on the relay, update/resolve take the original signed identity explicitly and list is the owner-side view.

File: crates/buzz-db/src/task.rs [hunk 10]
```diff
@@ -0,0 +1,700 @@
...
+            .await?;
+            if result.rows_affected() == 1 {
+                return Ok(TaskProjectionOutcome::Inserted);
+            }
+
+            let existing = select_projection_identity(tx, community_id, task.task_id).await?;
+            if let Some(existing) = existing {
+                validate_identity(&existing, task)?;
+                if existing.source_version >= payload.source_version {
+                    return Ok(TaskProjectionOutcome::Stale);
```
[the remark this answers is not in the document]
Comment (FurnitAI) [reply crates/buzz-db/src/task.rs]: Addressed in d84759362 (fix(tasks): reject conflicting duplicate task requests). An equal-version kind-44300 request whose payload differs from the stored v1 projection row now returns an error and the transaction rolls back the event insert, so no conflicting equal-version competitor reaches the public stream; only a byte-identical duplicate remains Stale. Locked in by a new task_projection test.
